Adding Tracking Catalog doc (#156)
snowplow/documentation: docs/data-discovery/tracking-catalog/index.md

---

> title: "Data Discovery through the Tracking Catalog"

Reviewer: implemented but to be discussed

date: "2022-10-26"
sidebar_position: 0
---

> ## Introducing the Tracking Catalog

Reviewer: implemented but to be discussed

> 

Reviewer: done

Discovery and understanding of the data your organisation is tracking through Snowplow is essential to enable teams to unlock real value from that data.
We know that in many cases, Data Analysts, Data Scientists and Product Managers struggle to find out what is being tracked, and what the data means. They normally have to rely on tribal knowledge and source this information directly with the front end developers who instrumented the tracking, or raise a request to the Data Team.

> However, through the Tracking Catalog, all users can have a holistic view of all their tracking in a always up-to-date and easy to consume view.

Reviewer 2: ```suggestion
However, through the Tracking Catalog, all users can have a holistic view of all their tracking in an always up-to-date and easy to consume view.
```

Reviewer: done

> However, through the Tracking Catalog, all users can have a holistic view of all their tracking in a always up-to-date and easy to consume view.

Reviewer: done

* * *

> ## Browsing the Tracking Catalog

Reviewer 2: Can we make this more goal-oriented? Like “Understanding what data is tracked”? That would help a lot with search and SEO

Reviewer: done

> **_NOTE:_** Please note that the Tracking Catalog is available only for BDP Customers on selected Tiers. If you're unsure about your eligibility, please get in touch with your Customer Success Manager.

Reviewer 2: This note should be the first thing on the page.

Reviewer: done

> Once logged in Snowplow Console, you will see a new link on the left hand side navigation, called Tracking Catalog. 

Reviewer 2: I.e. remove “new”

> Once logged in Snowplow Console, you will see a new link on the left hand side navigation, called Tracking Catalog. 

Reviewer: done

> ![](images/Tracking-Catalog-Nav.png)

Reviewer: done

Once you navigate to it, you will be presented with a view showing all the Events and Entities that have been tracked your Snowplow production pipelines in the past 90 days.

![](images/TC-Landing-page.png)

> If you select a particular Event, you can see the details of the Events, what properties are tracked for this event, described in plain English therefore easily consumable by a non-technical audience. You can also see any validation criteria associated with this event. The catalog will also display all canonical properties which are also tracked out of the box, when you use Snowplow trackers.

Reviewer 2: > If you select a particular Event, you can see the details of the Events

I am not sure I understand what this means. Should it be “Event” (singular) the second time as well?

Reviewer: done

> If you select a particular Event, you can see the details of the Events, what properties are tracked for this event, described in plain English therefore easily consumable by a non-technical audience. You can also see any validation criteria associated with this event. The catalog will also display all canonical properties which are also tracked out of the box, when you use Snowplow trackers.

Reviewer: - Capitalization ✅
- Links ✅
- Synchronization of "fields" and "properties" ❌

> If you select a particular Event, you can see the details of the Events, what properties are tracked for this event, described in plain English therefore easily consumable by a non-technical audience. You can also see any validation criteria associated with this event. The catalog will also display all canonical properties which are also tracked out of the box, when you use Snowplow trackers.

Reviewer: We would prefer to leave "properties" like it is, as the feature in Console and the marketing material uses this terminology and later on align the docs on a unified name.

![](images/event-details.png)

> ![](images/canonical-properties.png)

Reviewer 3: Just noticed one other little thing, the back to back images are a little tricky to look at when looking at the doc page, I'd perhaps suggest moving the sentence from the above paragraph to be in between these images:

> The catalog will also display all canonical properties which are also tracked out of the box, when you use Snowplow trackers.

Reviewer: done

If you scroll down the page, you will see an Event map. 

> This map display what entities are being tracked with this event. This information is really valuable to understand the connections between the data.

Reviewer 2: ```suggestion
This map displays what entities are being tracked with the event. This information is really valuable to understand the connections between the data.
```

Reviewer: done

> It also helps data consumers to make sense of how data is structured in the data warehouse and make it easier for them to understand how to query it.

Reviewer 2: ```suggestion
It also helps data consumers to make sense of how data is structured in the data warehouse and makes it easier for them to understand how to query it.
```
Does it?

Reviewer: Implemented, but to be answered

Reviewer: In our opinion it does.

![](images/event-map.png)

> Similarly, you can navigate to an Entity, by simply selecting one from the map or the list view, and explore details of it structure and what Events it is being tracked with.

Reviewer 2: ```suggestion
Similarly, you can navigate to an Entity by selecting one from the map or the list view. This shows you how it’s structured and what Events include it.
```

Reviewer: modified

Reviewer: done

> For both Events and Entity, the Tracking Catalog also displays a log of version changes to help users understand how the data is evolving through time.

Reviewer: done

![](images/versions.png)

> * * *

Reviewer: We would prefer to keep them in.

## Discovery through the Tracking Catalog

> Often, data consumers only have high level information to guide their data discovery process, and are not aware of the exact name of the Data Structure used for tracking certain data.

Reviewer: Please can you let it slide for now, we would rather give the whole thing an overhaul with this mindset than updating a single paragraph but this feature is being rolled out at the moment.

> This is why the Tracking Catalog search function will map results of a search to various potential properties. 

Reviewer: done

> When a search is carried out, the Tracking Catalog will try and find results matching the searched text in both the Data Structure name, its description, its properties's names and potential enumeration of values.

Reviewer 2: ```suggestion
When you search through the Tracking Catalog, it will look for matches in the Data Structure names, descriptions and fields names. In addition, for fields with a fixed set of allowed values (e.g. a `stage` field with values `checkout` and `payment`) Tracking Catalog will look for matches in the values.
```

Reviewer 2: Editing for clarity

Reviewer: implemented with field -> property replacement

> If a match is found, it will highlight the reason behind the match.

Reviewer 2: ```suggestion
You can find the reason behind the match on the right.
```

Reviewer: done

![](images/search.png)

> The Tracking Catalog will provide your team with easy and self serve access to this data, creating greater transparency and autonomy. 

Reviewer 2: I think placed here at the bottom of the page this paragraph loses its punch. I would either put it at the top, in the “Introduction” section, or remove entirely (on the assumption that the benefits are self-evident to anyone who’s read the page).

Reviewer: We would still like to keep it here for now and maybe do a second round later.
